libfive/stdlib/stdlib_impl.cpp

```diff
@@ -0,0 +1,1379 @@
+/*
+libfive: a CAD kernel for modeling with implicit functions
+
+Copyright (C) 2021  Matt Keeter
+
+This Source Code Form is subject to the terms of the Mozilla Public
+License, v. 2.0. If a copy of the MPL was not distributed with this file,
+You can obtain one at http://mozilla.org/MPL/2.0/.
+*/
+#include "stdlib_impl.hpp"
+
+using namespace libfive;
+
+#define LIBFIVE_DEFINE_XYZ() const auto x = Tree::X(); (void)x; \
+                             const auto y = Tree::Y(); (void)y; \
+                             const auto z = Tree::Z(); (void)z; ;
+////////////////////////////////////////////////////////////////////////////////
+// Operator overloads for C vecs
+TreeVec2 operator+(const TreeVec2& a, const TreeVec2& b) {
+    return TreeVec2{a.x + b.x, a.y + b.y};
+}
+TreeVec2 operator-(const TreeVec2& a, const TreeVec2& b) {
+    return TreeVec2{a.x - b.x, a.y - b.y};
+}
```

Comment:
Maybe for completeness also add a negation operator for TreeVec2
```suggestion
}
TreeVec2 operator-(const TreeVec2& a) {
    return TreeVec2{-a.x, -a.y};
}
```